bgpd/bgp_vty.c

```diff
@@ -11324,15 +11329,19 @@ static int bgp_show_route_leak_vty(struct vty *vty, const char *name,
 				json_object_array_add(json_import_vrfs,
 						json_object_new_string(vname));
 
+			json_object_object_add(json, "importFromVrfs",
+						       json_import_vrfs);
 			dir = BGP_VPN_POLICY_DIR_FROMVPN;
-			ecom_str = ecommunity_ecom2str(
+			if (bgp->vpn_policy[afi].rtlist[dir]) {
+				ecom_str = ecommunity_ecom2str(
 					bgp->vpn_policy[afi].rtlist[dir],
 					ECOMMUNITY_FORMAT_ROUTE_MAP, 0);
-			json_object_object_add(json, "importFromVrfs",
-					       json_import_vrfs);
```

Comment:
looks like importFromVrfs is dropped from the json output, is this intentional? If so, why?

Reply:
@louberger  It is not dropped, with the change it is at line 11332.

Reply:
sorry, i missed that. Thanks.  Any chance to get a topotest covering this case?

Reply:
We will have to take that as a separate item. Need to see what's covered for vrf leak and add new test case covering deletion of non-default vrf.